### PR TITLE
Define adjoint for softplus

### PR DESCRIPTION
**Pull request by AStupidBear**
fix https://github.com/FluxML/Flux.jl/issues/876

### CODE REVIEW

File: src/lib/nnlib.jl [hunk 1]
```diff
@@ -10,6 +10,11 @@ end
     return y, Δ -> (Δ * y * (1 - y),)
 end
 
+@adjoint function softplus(x::Real)
+    y = softplus(x)
+    return y, Δ -> (Δ * σ(*),)
```
Comment (dhairyagandhi96) [inline src/lib/nnlib.jl:15]: This doesn't look right. Maybe it should be 
```suggestion
    return y, Δ -> (Δ * σ(x),)
```

Comment (AStupidBear) [reply src/lib/nnlib.jl]: Sorry, a stupid typo. Now fixed.

**Comment by CarloLucibello** [comment]
tests are failing due to some tiny numerical error here, @AStupidBear can we fix that or make tests more robust?

**Comment by devmotion** [comment]
Couldn't/shouldn't the custom adjoint be defined in NNlib using ChainRulesCore such that other AD frameworks could use it as well?

**Comment by dhairyagandhi96** [comment]
We can move over the adjoints to frules but that would need to be done for a lot more than functions than just this. We can have this in now and move the adjoints as a different issue if that works

**Comment by devmotion** [comment]
Sure, that could probably done for other definitions as well. I was wondering mainly because this PR adds a new definition, so one could avoid having to move it around later by adding it to NNlib right away.

**Comment by CarloLucibello** [comment]
this should be filed to NNlib.jl

**Pull request closed without merging**